src/System.Private.ServiceModel/src/System/ServiceModel/Channels/Message.cs

```diff
@@ -451,20 +451,24 @@ public override string ToString()
                 return base.ToString();
             }
 
-            StringWriter stringWriter = new StringWriter(CultureInfo.InvariantCulture);
+            
             XmlWriterSettings settings = new XmlWriterSettings();
             settings.Indent = true;
-            XmlWriter textWriter = XmlWriter.Create(stringWriter, settings);
-            XmlDictionaryWriter writer = XmlDictionaryWriter.CreateDictionaryWriter(textWriter);
-            try
-            {
-                ToString(writer);
-                writer.Flush();
-                return stringWriter.ToString();
-            }
-            catch (XmlException e)
+
+            using (StringWriter stringWriter = new StringWriter(CultureInfo.InvariantCulture))
+            using (XmlWriter textWriter = XmlWriter.Create(stringWriter, settings))
```

Comment:
I think it would be better to use curly braces for each nested statement for clarity -- even though it will indent quite a bit.